- module: map
  mode: [online, logging_simulation, planning_simulation]
  type: launch
  args:
    node_name_suffix: vector_map
    topic: /map/vector_map
    topic_type: autoware_auto_mapping_msgs/msg/HADMapBin
    best_effort: false
    transient_local: true
    warn_rate: 0.0
    error_rate: 0.0
    timeout: 0.0

- module: map
  mode: [online, logging_simulation]
  type: launch
  args:
    node_name_suffix: pointcloud_map
    topic: /map/pointcloud_map
    topic_type: sensor_msgs/msg/PointCloud2
    best_effort: false
    transient_local: true
    warn_rate: 0.0
    error_rate: 0.0
    timeout: 0.0

- module: localization
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: initialpose3d
    topic: /initialpose3d
    topic_type: geometry_msgs/msg/PoseWithCovarianceStamped
    best_effort: false
    transient_local: false
    warn_rate: 0.0
    error_rate: 0.0
    timeout: 0.0

- module: localization
  mode: [online, logging_simulation]
  type: autonomous
  args:
    node_name_suffix: pose_twist_fusion_filter_pose
    topic: /localization/pose_twist_fusion_filter/pose
    topic_type: geometry_msgs/msg/PoseStamped
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: localization
  mode: [online, logging_simulation]
  type: autonomous
  args:
    node_name_suffix: pose_estimator_pose
    topic: /localization/pose_estimator/pose_with_covariance
    topic_type: geometry_msgs/msg/PoseWithCovarianceStamped
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: perception
  mode: [online, logging_simulation]
  type: launch
  args:
    node_name_suffix: obstacle_segmentation_pointcloud
    topic: /perception/obstacle_segmentation/pointcloud
    topic_type: sensor_msgs/msg/PointCloud2
    best_effort: true
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: perception
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: object_recognition_objects
    topic: /perception/object_recognition/objects
    topic_type: autoware_auto_perception_msgs/msg/PredictedObjects
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: perception
  mode: [online, logging_simulation]
  type: autonomous
  args:
    node_name_suffix: traffic_light_recognition_traffic_signals
    topic: /perception/traffic_light_recognition/traffic_signals
    topic_type: autoware_perception_msgs/msg/TrafficSignalArray
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: planning
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: mission_planning_route
    topic: /planning/mission_planning/route
    topic_type: autoware_planning_msgs/msg/LaneletRoute
    best_effort: false
    transient_local: true
    warn_rate: 0.0
    error_rate: 0.0
    timeout: 0.0

- module: planning
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: scenario_planning_trajectory
    topic: /planning/scenario_planning/trajectory
    topic_type: autoware_auto_planning_msgs/msg/Trajectory
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: control
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: trajectory_follower_control_cmd
    topic: /control/trajectory_follower/control_cmd
    topic_type: autoware_auto_control_msgs/msg/AckermannControlCommand
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: control
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: control_command_control_cmd
    topic: /control/command/control_cmd
    topic_type: autoware_auto_control_msgs/msg/AckermannControlCommand
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: vehicle
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: vehicle_status_velocity_status
    topic: /vehicle/status/velocity_status
    topic_type: autoware_auto_vehicle_msgs/msg/VelocityReport
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: vehicle
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: vehicle_status_steering_status
    topic: /vehicle/status/steering_status
    topic_type: autoware_auto_vehicle_msgs/msg/SteeringReport
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: system
  mode: [online, logging_simulation, planning_simulation]
  type: launch
  args:
    node_name_suffix: system_emergency_control_cmd
    topic: /system/emergency/control_cmd
    topic_type: autoware_auto_control_msgs/msg/AckermannControlCommand
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0

- module: localization
  mode: [online, logging_simulation, planning_simulation]
  type: autonomous
  args:
    node_name_suffix: transform_map_to_base_link
    topic: /tf
    frame_id: map
    child_frame_id: base_link
    best_effort: false
    transient_local: false
    warn_rate: 5.0
    error_rate: 1.0
    timeout: 1.0
